src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,160 @@
+      const valueB = b[sortField];
+
+      if (typeof valueA === 'string' && typeof valueB === 'string') {
+        return orderName
+          ? valueB.localeCompare(valueA)
+          : valueA.localeCompare(valueB);
+      }
+
+      if (typeof valueA === 'number' && typeof valueB === 'number') {
+        return orderName ? valueB - valueA : valueA - valueB;
+      }
+
+      return 0;
+    });
+  }
+
+  const sortedPeople = getSortedPeople(people, sort, order);
+
+  function getVisiblePeople(
+    folk: Person[],
+    gender: string,
+    inputQuery: string,
+    centuryParams: string[],
+  ) {
+    let visible = [...folk];
+
+    if (gender) {
+      visible = visible.filter(person => person.sex === sex);
+    }
+
+    if (inputQuery) {
+      visible = visible.filter(
+        person =>
+          person.name.toLowerCase().includes(inputQuery.toLowerCase()) ||
+          person.fatherName?.toLowerCase().includes(inputQuery.toLowerCase()) ||
+          person.motherName?.toLowerCase().includes(inputQuery.toLowerCase()),
+      );
+    }
+
+    if (centuryParams.length > 0) {
+      visible = visible.filter(person => {
+        const century = Math.ceil(person.born / 100).toString();
+
+        return centuryParams.includes(century);
+      });
+    }
+
+    return visible;
+  }
+
+  const visiblePeople = getVisiblePeople(sortedPeople, sex, query, centuries);
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
       <div className="block">
+        {loading && <Loader />}
+
         <div className="columns is-desktop is-flex-direction-row-reverse">
-          <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
-          </div>
+          {!loading && (
+            <div className="column is-7-tablet is-narrow-desktop">
+              <PeopleFilters
+                searchParams={searchParams}
+                setSearchParams={setSearchParams}
+                sex={sex}
+                query={query}
+                centuries={centuries}
+              />
+            </div>
+          )}
 
           <div className="column">
             <div className="box table-container">
-              <Loader />
-
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+              {errorMessage && (
+                <p data-cy="peopleLoadingError" className="has-text-danger">
+                  {errorMessage}
+                </p>
+              )}
 
-              <p data-cy="noPeopleMessage">There are no people on the server</p>
+              {people.length === 0 && !loading && !errorMessage && (
+                <p data-cy="noPeopleMessage">
+                  There are no people on the server
+                </p>
+              )}
 
-              <p>There are no people matching the current search criteria</p>
+              {visiblePeople.length === 0 && (
+                <p>There are no people matching the current search criteria</p>
+              )}
```

Comment:
Ensure that the message about no matching people is conditionally rendered only when no visible people match the search criteria. Consider wrapping this message in a conditional check that includes `!loading` and `!errorMessage`.